### PR TITLE
act helper scripts

### PR DESCRIPTION
**Pull request by Alb0t**

### CODE REVIEW

**Comment by Alb0t** [comment]
@gooding470 the new ACT still requires you to build out a bunch of silly config files manually. I updated the script to allow all (i think) options to be configurable and coded in the defaults. I believe it's still useful

**Comment by Alb0t** [comment]
I'm hoping as the setup instructions we can give an alternative of running yum|apt/clone/make as simply `curl https://github.com/aerospike/act/ala/act_install.sh | sudo bash` then you can build out the configs using the helper script, then have the bash script run all the tests... makes life better.

**Comment by gooding470** [comment]
Hi Albert,

I haven't actually looked at the details (ok ... anything) yet, but in case it helps, config files don't need to contain default values, there are only 4 items that need to be there (device-names, test-duration-sec, read-reqs-per-sec and write-reqs-per-sec).  If your script is allowing you to specify non-defaults for other items, it only needs to add the thing(s) specified.  But maybe you already do all this.  Also, whatever you're doing for the storage config, if useful, we should also do for the index config.

When I get the chance I'll actually look at what you've done.  We may of course tweak it up a bit if we think it needs it.

-- Andy

**Comment by Alb0t** [comment]
Thanks Andy. I coded in the defaults because I didn't want to make a bunch of case statements and I'm a horrible ruby scripter. Adding in the defaults made it easy. I'm sure a few tweaks will be required, and it will need to be changed a bit to work with index, but I think you should be able to get the gist of what I'm trying to do. Let me know if a quick chat might help too!

File: ala/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+
+### Overview
+------------
+1. act_install.sh - install act on deb/el systems
+    * Usage: `sudo bash act_install.sh`
+2. act_make_configs.rb - create config files using ruby.
+    * This script will create config files for storage testing (not index) based off the passed parameters to run act against. By default it will create 1 actconfig for every 10X increments. By default it will create files for 10X-120X so you will have 12 files generated. 10x,20x,..etc...120x.
+    * You will need to have ruby installed and run 'gem install optimist' for this to work. 
+    * Usage: `apt install ruby || yum install ruby; gem install optimist; ruby act_make_configs.rb -d /dev/drive1p1,/dev/drive2p1`,etc... for help run without args or `ruby act_make_configs.rb -h`
```
Comment (kportertx) [inline ala/README.md]: We are already using Python for the analysis, we would prefer to not introduce another scripting language and dependencies.

File: ala/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+3. runact.sh - run through a series of act configs and save the output.
+    * This will start from 120x tests and continue down in steps of 10 to create act test results from 120 -> 10x based on the configs generated in act_make_configs.rb.
+    * Usage: `bash runact.sh` default: 120x-10x tests. To modify the start/end of the act tests (must have configs present) use `bash runact.sh upto downto` ex.. `bash runact.sh 120 20` will run act tests from 120x to 20x.
```
Comment (kportertx) [inline ala/README.md]: I believe [run_act_storage.sh](https://github.com/aerospike/act/pull/35/files?utf8=%E2%9C%93&diff=unified#diff-b26ae2cc341fdf43edce9bae64878365R29) hard codes this to 12.

File: ala/act_install [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+#Work with EL/Deb
+apt-get -y install make gcc libc6-dev libssl-dev zlib1g-dev git || yum install -y make gcc openssl-devel sysstat git 
+cd /opt
+#git clone https://github.com/brendangregg/perf-tools.git
+git clone https://github.com/aerospike/act.git
+#export PATH="$PATH:/opt/perf-tools"
+cd /opt/act
+make
+#make -f Makesalt #not required with latest dev branch
```
Comment (kportertx) [inline ala/act_install]: Could we remove commented out commands.

File: ala/act_make_configs.rb [hunk 4]
```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env ruby
+require 'optimist' #for options parsing
+
+opts = Optimist::options do
+    opt :devices, "REQUIRED: List of devices to test. Ex: /dev/nvme3n1p1,/dev/nvme1n1p4", :default => ""
+    opt :directory, "Directory to save config files to. (No trailing /, directory must exist)", :default => "/opt"
+    opt :x, "How many 'X' configs you want to generate. At the default of 120, it will generate 120 actconfigs.", :default => 120
```
Comment (kportertx) [inline ala/act_make_configs.rb]: You mentioned in the [readme](https://github.com/aerospike/act/pull/35/files?utf8=%E2%9C%93&diff=unified#diff-072e92c2457c780c24651dab0c26032dR9) that you intend to only generate 12 configs by default.

File: ala/act_make_configs.rb [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    opt :report_interval_sec, "How often to report histograms back for logging (seconds).", :default => 1
+    banner <<-EOS
+
+    Reference https://github.com/aerospike/act for more detailed explanation of ACT parameters.
+
+    EOS
+end
+Optimist::educate if(opts[:devices]=="") #print help if no devices passed, or if -h passed.
+
+raise "Direcotry: #{opts[:directory]} does not exist." if(!File.directory?(opts[:directory]))
```
Comment (kportertx) [inline ala/act_make_configs.rb]: "Directory"

File: ala/run_act_storage.sh [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+if [ ! -d $configfolder ]
+then
+    echo "ACT Config Folder: ${configfolder} does not exist. Exiting."
+fi
+
+echo "testing ${upto}X - ${downto}X configs in $configfolder"
+
+while `pidof actprep>/dev/null`; do echo "waiting for actprep: $(ps aux|grep -v grep|grep actprep)";sleep 60;done;sleep 60
+for n in {12..1}; do
```
Comment (kportertx) [inline ala/run_act_storage.sh]: Probably wanted "{$upto..$downto}".

**Comment by kportertx** [comment]
Also, what is 'ala'?

File: ala/README.md (hunk 1 above)
Comment (Alb0t) [reply ala/README.md]: I dont like python though. Please re-write in python :P

**Comment by Alb0t** [comment]
> Also, what is 'ala'?

my initials. I was pretty sure you guys wouldn't want to merge this, so just stuffed it into a folder so you could get an idea of what I'm trying to do here. It'd be better if ACT didn't require a ton of config files to use easily though (hint hint)

**Pull request closed without merging**